extensions/notebook/src/test/book/githubRemoteBook.test.ts

```diff
@@ -0,0 +1,117 @@
+		let releaseURL = vscode.Uri.parse('https://api.github.com/repos/microsoft/test/releases/v1');
+		let asset : IAsset = {
+			name: 'CU-1.0-EN.zip',
+			book: 'CU',
+			version: '1.0',
+			language: 'EN',
+			format: 'zip',
+			url: vscode.Uri.parse('https://api.github.com/repos/microsoft/test/releases/v1/assets/1'),
+			browserDownloadUrl: vscode.Uri.parse('https://github.com/microsoft/test/releases/download/v1/CU-1.0-EN.zip'),
+		}
+		let remoteLocation = loc.onGitHub;
+		controller.setRemoteBook(releaseURL, remoteLocation, asset);
+		should(controller.model.remoteBook).not.null();
+		should(controller.model.remoteBook instanceof GitHubRemoteBook).be.true;
+		let book = model.remoteBook as GitHubRemoteBook;
+		should(book.asset.browserDownloadUrl.toString(false)).equal('https://github.com/microsoft/test/releases/download/v1/CU-1.0-EN.zip');
+	});
+
+	it('Verify set local path is called when creating a GitHub Remote Book', async function (): Promise<void> {
+		let releaseURL = vscode.Uri.parse('https://api.github.com/repos/microsoft/test/releases/v1');
+		let asset : IAsset = {
+			name: 'CU-1.0-EN.zip',
+			book: 'CU',
+			version: '1.0',
+			language: 'EN',
+			format: 'zip',
+			url: vscode.Uri.parse('https://api.github.com/repos/microsoft/test/releases/v1/assets/1'),
+			browserDownloadUrl: vscode.Uri.parse('https://github.com/microsoft/test/releases/download/v1/CU-1.0-EN.zip'),
+		}
+		let remoteLocation = loc.onGitHub;
+		const createCopySpy = sinon.spy(GitHubRemoteBook.prototype, 'createLocalCopy');
+		const setPathSpy = sinon.spy(RemoteBook.prototype, 'setLocalPath');
+		controller.setRemoteBook(releaseURL, remoteLocation, asset);
+		should(createCopySpy.calledOnce).be.true;
+		should(setPathSpy.calledOnce).be.true;
+	});
+
+	it('Should download contents from Github', async function (): Promise<void> {
+		let releaseURL = vscode.Uri.parse('https://api.github.com/repos/microsoft/test/releases/v1');
+		let asset : IAsset = {
+			name: 'CU-1.0-EN.zip',
+			book: 'CU',
+			version: '1.0',
+			language: 'EN',
+			format: 'zip',
+			url: vscode.Uri.parse('https://api.github.com/repos/microsoft/test/releases/v1/assets/1'),
+			browserDownloadUrl: vscode.Uri.parse('https://github.com/microsoft/test/releases/download/v1/CU-1.0-EN.zip'),
+		}
+		let remoteLocation = loc.onGitHub;
+		controller.setRemoteBook(releaseURL, remoteLocation, asset);
+
+		model.remoteBook.localPath = vscode.Uri.file(os.tmpdir());
+		let setPathStub = sinon.stub(GitHubRemoteBook.prototype, 'setLocalPath');
+		setPathStub.callsFake(function() {
+			console.log(`Downloading book in ${model.remoteBook.localPath}`);
+		})
+		const setExtractSpy = sinon.spy(GitHubRemoteBook.prototype, 'extractFiles');
+		nock('https://github.com')
+				.persist()
+				.get('/microsoft/test/releases/download/v1/CU-1.0-EN.zip')
+				.replyWithFile(200, __filename);
+		await should(model.remoteBook.createLocalCopy()).be.fulfilled();
```

Comment:
This is fine - but it's also not necessary if you just await the function call since it'll fail the test if it throws regardless.